src/agent/src/rpc.rs

```diff
@@ -1542,18 +1585,15 @@ fn do_copy_file(req: &CopyFileRequest) -> Result<()> {
         PathBuf::from("/")
     };
 
-    fs::create_dir_all(dir.to_str().unwrap()).or_else(|e| {
+    fs::create_dir_all(dir.clone()).or_else(|e| {
```

Comment:
You shouldn't need to `clone()` here. No idea what the `to_str()` was about in the first place, either.  AFAICT just passing `&dir` should be fine.

Reply:
You still have `clone()`.  Just `fs::create_dir_all(&dir)` should do the trick.

Reply:
Fixed.